sympy/vector/vector.py

```diff
@@ -418,19 +418,22 @@ def _div_helper(one, other):
 
 def get_postprocessor(cls):
     def _postprocessor(expr):
-        vec_class = {Add: VectorAdd}[cls]
-        vectors = []
-        for term in expr.args:
-            if isinstance(term.kind, VectorKind):
-                vectors.append(term)
+        vec_class = {Add: VectorAdd, Mul: VectorMul, Pow: None}[cls]
 
         if vec_class == VectorAdd:
-            return VectorAdd(*vectors).doit(deep=False)
+            return VectorAdd(*expr.args).doit(deep=False)
+        elif vec_class == VectorMul:
+            return VectorMul(*expr.args).doit(deep=False)
+        elif vec_class is None:
+            if isinstance(expr.args[0], Vector):
+                raise TypeError("Power operation is not supported for vectors")
     return _postprocessor
 
 
 Basic._constructor_postprocessor_mapping[Vector] = {
     "Add": [get_postprocessor(Add)],
+    "Mul": [get_postprocessor(Mul)],
+    "Pow": [get_postprocessor(Pow)],
 }
```

Comment:
The `get_postprocessor` function seems like a pointless complication. Better just to have
```
def _add_vector_processor(expr: Add) -> Vector:
    return VectorAdd(*expr.args).doit(deep=False)
```